### PR TITLE
bragi: error handling

### PR DESCRIPTION
**Pull request by kadhikari**
Catch the error with following messages:
- "Elasticsearch Connection Error"
- "Index not found"

### CODE REVIEW

File: libs/bragi/src/query.rs [hunk 1]
```diff
@@ -214,8 +214,21 @@ fn build_query(q: &str,
...
-    !index.is_empty() && client.open_index(&index).is_ok()
+fn is_existing_index(client: &mut rs_es::Client,
+                     index: &str)
+                     -> Result<bool, rs_es::error::EsError> {
+    if index.is_empty() {
+        return Ok(false);
+    }
+    match client.open_index(&index) {
+        Err(::rs_es::error::EsError::EsServerError(_)) => {
+            Err(rs_es::error::EsError::EsServerError("Index not found ".to_string()))
```
Comment (pbougue) [inline libs/bragi/src/query.rs]: you probably can reuse original error and do a message like `format!("Index not found: {}", e.to_string())`
probably not working, but you have the idea (same below).

Comment (pbougue) [inline libs/bragi/src/query.rs]: And you have an extra space (in case you don't change message).
Same below.

File: libs/bragi/src/query.rs [hunk 2]
```diff
@@ -214,8 +214,17 @@ fn build_query(q: &str,
...
-fn is_existing_index(client: &mut rs_es::Client, index: &str) -> bool {
-    !index.is_empty() && client.open_index(&index).is_ok()
+fn is_existing_index(client: &mut rs_es::Client,
+                     index: &str)
+                     -> Result<bool, rs_es::error::EsError> {
+    if index.is_empty() {
+        return Ok(false);
+    }
+    match client.open_index(&index) {
+        Err(::rs_es::error::EsError::EsServerError(_)) => Ok(false),
```
Comment (antoine-de) [inline libs/bragi/src/query.rs]: I don't understand why this is valid. We want an error if no ES can be found no ?

Comment (antoine-de) [inline libs/bragi/src/query.rs]: Ok I get what this error is, but I think a comment explaing what is this error would be great

**Pull request merged**